### PR TITLE
fix: prevent pkgname or pkgbase from having capital letters

### LINKED ISSUE #78: makedeb should abort if `pkgbase` or `pkgname` contains a capital letter
### Issue

See https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-source.

### Installation source

APT Repository

### Installed release type

Alpha

### Installed makedeb version

8.6.0-1

### Installed makepkg version

N/A

### PR DESCRIPTION
**Pull request by Elsie19**
Fixes #78

### CODE REVIEW

File: src/makedeb/main.sh [hunk 1]
```diff
@@ -105,6 +105,16 @@ generate_prefix_fields
...
+if [[ "${pkgbase}" =~ [A-Z] ]]; then
+  error "'pkgbase' contains capital letters"
+  exit 1
+fi
+
+if [[ "${pkgname}" =~ [A-Z] ]]; then
+  error "'pkgname' contains capital letters"
+  exit 1
+fi
+
```
Comment (hwittenborn) [inline src/makedeb/main.sh]: This should be placed in `src/makepkg/functions/lint_pkgbuild/pkgname.sh`, as well any future linting checks.

I think there's some still in makedeb's code, but I'm wanting them to stay in makepkg if at all possible.

**Review by hwittenborn — CHANGES_REQUESTED** [review]

**Review by lpuv — APPROVED** [review]
LGTM

File: src/makepkg/functions/lint_pkgbuild/pkgbase.sh [hunk 2]
```diff
@@ -34,6 +34,10 @@ lint_pkgbase() {
 	if [[ -z $pkgbase ]]; then
 		return 0
 	fi
+	if [[ "${pkgbase}" =~ [A-Z] ]]; then
+  	error "'pkgbase' contains capital letters"
+  	return 1
+	fi
```
Comment (hwittenborn) [inline src/makepkg/functions/lint_pkgbuild/pkgbase.sh]: Can you add a newline above this `if` statement?

**Review by hwittenborn — CHANGES_REQUESTED** [review]

File: src/makepkg/functions/lint_pkgbuild/pkgname.sh [hunk 3]
```diff
@@ -88,6 +88,11 @@ lint_pkgname() {
 		done
 	fi
 
+	if [[ "${pkgname}" =~ [A-Z] ]]; then
+		error "'pkgname' contains capital letters"
+		ret=1
+	fi
```
Comment (hwittenborn) [inline src/makepkg/functions/lint_pkgbuild/pkgname.sh]: This and the other change in `pkgbase.sh` can both just be put in `lint_one_pkgname`, right?

**Review by hwittenborn — CHANGES_REQUESTED** [review]

File: src/makepkg/functions/lint_pkgbuild/pkgname.sh [hunk 4]
```diff
@@ -73,6 +73,16 @@ lint_one_pkgname() {
...
+
+	if [[ "${pkgbase}" =~ [A-Z] ]]; then
+  	error "'pkgbase' contains capital letters"
+  	return 1
+	fi
+
```
Comment (hwittenborn) [inline src/makepkg/functions/lint_pkgbuild/pkgname.sh]: This should be the last thing, but you could be able to put these under one `if` statement but just checking the `${type}` and `${name}` variables as with the statements above it.

**Review by hwittenborn — CHANGES_REQUESTED** [review]

**Comment by hwittenborn** [comment]
Can you merge in the upstream changes? It should fix that downgrading issue that appears during unit tests.

**Review by hwittenborn — APPROVED** [review]

**Review by hwittenborn — APPROVED** [review]
A-OK to me

**Review by lpuv — APPROVED** [review]
LGTM

**Pull request merged**